### PR TITLE
:robot: feat: WhatsApp bot inteprets outgoing LocationMessageBlock

### LINKED ISSUE #208: 🤖 📍  Enable the bot engine to interpret an Outgoing LocationMessageBlock to a Whatsapp Message
## User stories

As an end user I should be able to receive a location message block on whatsapp as a location message.
---

Quick links: [Description](#description)  | [Resources](#resources)

---

### Description
---

**NOTE: You will also need to write unit tests for the specific function implemented. You can refer to this [PR](https://github.com/italanta/elewa/pull/222/files) to see how I have written them and where they are located. We are using jest.**

**The tests will automatically be run when you create a pr. So make sure your code passes the tests before requesting review.**

**Feel free to reach out on slack if you need more info.**

---

Under **functions/bot-engine/whatsapp/io** we have a `WhatsappOutgoingMessageParser` which is reponsible for intepreting incoming user messages from whatsapp to a standardized message format. Please go through the comments carefully and refer to the documentation below

They only class you need for this is the `WhatsappOutgoingMessageParser` and the parent class.

You can review how other block types have been done.

### Resources:

* Setting up a whatsapp developer account for testing: Read this [Readme](https://github.com/italanta/elewa/blob/main/libs/functions/bot-engine/whatsapp/README.md)
* Whatsapp Send Messages Documentation : [Payload Examples](https://developers.facebook.com/docs/whatsapp/cloud-api/guides/send-messages)

### Acceptance Criteria

1. You must show clear conversation of the outgoing message message to a standardized message with comments
2. The message should be received on whatsapp as a location

### PR DESCRIPTION
**Pull request by kelvinmburu**
# Description

The changes for this PR address issue #208 which relates to the backend (WhatsApp bot engine)

This PR fixes #208 .

## Type of Change

- [x] New feature (non-breaking change which adds functionality)

# Screenshot

# How Has This Been Tested?

- [x] Test clear conversation of the outgoing message to a standardized message with comments
- [ ] The message should be received on WhatsApp as a location (Yet to test)

# Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works

### CODE REVIEW

**Comment by ReaganCn** [comment]
Good Morning Kelvin,

Good work so far but for backend tasks you will also need to write unit tests for the specific function implemented. You can refer to this [PR](https://github.com/italanta/elewa/pull/222/files) to see how I have written them and where they are located. I know this was not included in the initial issue description but will add it.

Thank you.

**Comment by kelvinmburu** [comment]
Thanks. Let me add the test

File: libs/features/convs-mgr/stories/home/src/lib/components/story-list/story-list.component.html [hunk 1]
```diff
@@ -1,15 +1,11 @@
 <!-- edited the search -->
```
Comment (ReaganCn) [inline libs/features/convs-mgr/stories/home/src/lib/components/story-list/story-list.component.html:1]: I dont understand why we are editing this file and it is a backend issue

File: libs/features/convs-mgr/stories/home/src/lib/components/story-list/story-list.component.scss [hunk 2]
```diff
@@ -52,6 +52,7 @@
   margin-left: 24em;
```
Comment (ReaganCn) [inline libs/features/convs-mgr/stories/home/src/lib/components/story-list/story-list.component.scss:52]: Also this.

File: libs/features/convs-mgr/stories/home/src/lib/components/story-list/story-list.component.ts [hunk 3]
```diff
@@ -3,6 +3,10 @@ import { BehaviorSubject, combineLatest, map, Observable } from 'rxjs';
 import { StoriesStore } from '@app/state/convs-mgr/stories';
```
Comment (ReaganCn) [inline libs/features/convs-mgr/stories/home/src/lib/components/story-list/story-list.component.ts:3]: Also this.

**Review by ReaganCn — CHANGES_REQUESTED** [review]
Good work. But I dont understand why we are editing front end files and it is a backend issue. Also you need to write tests so that we can see if it works

File: libs/features/convs-mgr/stories/home/src/lib/components/story-list/story-list.component.html (hunk 1 above)
Comment (kelvinmburu) [reply libs/features/convs-mgr/stories/home/src/lib/components/story-list/story-list.component.html:1]: Hello @ReaganCn, There was a mixup when I was making the PR for this backend issue. The commit for issue #72 appeared on this PR which wasn't supposed to happen.

Comment (ReaganCn) [reply libs/features/convs-mgr/stories/home/src/lib/components/story-list/story-list.component.html:1]: It's okay. You can make the changes and push them

**Pull request closed without merging**